File: src/types/mcp-tool.ts
import { BaseEntity, TransportType } from './common';

// MCP Tool types
export interface MCPTool extends BaseEntity {
  name: string;
  description: string;
  enabled: boolean;
  endpoint_url: string;
  transport: TransportType;
  required_permissions?: string[];
  auth_headers?: Record<string, string>;
}

export interface CreateMCPToolRequest {
  name: string;
  description: string;
  enabled: boolean;
  endpoint_url: string;
  transport: TransportType;
  required_permissions?: string[];
  auth_headers?: Record<string, string>;
}

export interface UpdateMCPToolRequest {
  name?: string;
  description?: string;
  enabled?: boolean;
  endpoint_url?: string;
  transport?: TransportType;
  required_permissions?: string[];
  auth_headers?: Record<string, string>;
}

export interface MCPToolRBAC {
  mcp_tool_id: string;
  roles_access: string[];
  auth_header_type: 'API Key' | 'Bearer Token' | 'None';
  auth_header_value: string;
}

File: portal-be/app/models/mcp_tool.py
"""
MCP Tool model
"""
from sqlalchemy import Column, String, Boolean, Text, JSON
from .base import BaseModel


class MCPTool(BaseModel):
    __tablename__ = "mcp_tools"

    name = Column(String(255), nullable=False)
    description = Column(Text)
    enabled = Column(Boolean, default=True)
    endpoint_url = Column(String(500), nullable=False)
    transport = Column(String(100), default="Streamable HTTP")
    required_permissions = Column(JSON)
    auth_headers = Column(JSON)
